drivers/ieee802154/ieee802154_cc2520.c

```diff
@@ -971,135 +938,50 @@ static int power_on_and_setup(const struct device *dev)
+static int configure_gpios(const struct device *dev)
 {
-	struct cc2520_context *cc2520 = dev->data;
-	const struct device *gpio;
-
-	/* VREG_EN */
-	gpio = device_get_binding(DT_INST_GPIO_LABEL(0, vreg_en_gpios));
-	if (!gpio) {
-		return NULL;
-	}
-
-	cc2520->gpios[CC2520_GPIO_IDX_VREG_EN].pin = DT_INST_GPIO_PIN(0, vreg_en_gpios);
-	gpio_pin_configure(gpio, cc2520->gpios[CC2520_GPIO_IDX_VREG_EN].pin,
-			   GPIO_OUTPUT_LOW | DT_INST_GPIO_FLAGS(0, vreg_en_gpios));
-	cc2520->gpios[CC2520_GPIO_IDX_VREG_EN].dev = gpio;
-
-	/* RESET */
-	gpio = device_get_binding(DT_INST_GPIO_LABEL(0, reset_gpios));
-	if (!gpio) {
-		return NULL;
-	}
-
-	cc2520->gpios[CC2520_GPIO_IDX_RESET].pin = DT_INST_GPIO_PIN(0, reset_gpios);
-	gpio_pin_configure(gpio, cc2520->gpios[CC2520_GPIO_IDX_RESET].pin,
-			   GPIO_OUTPUT_LOW | DT_INST_GPIO_FLAGS(0, reset_gpios));
-	cc2520->gpios[CC2520_GPIO_IDX_RESET].dev = gpio;
-
-	/*FIFO */
-	gpio = device_get_binding(DT_INST_GPIO_LABEL(0, fifo_gpios));
-	if (!gpio) {
-		return NULL;
-	}
-
-	cc2520->gpios[CC2520_GPIO_IDX_FIFO].pin = DT_INST_GPIO_PIN(0, fifo_gpios);
-	gpio_pin_configure(gpio, cc2520->gpios[CC2520_GPIO_IDX_FIFO].pin,
-			   GPIO_INPUT | DT_INST_GPIO_FLAGS(0, fifo_gpios));
-	cc2520->gpios[CC2520_GPIO_IDX_FIFO].dev = gpio;
-
-	/* CCA */
-	gpio = device_get_binding(DT_INST_GPIO_LABEL(0, cca_gpios));
-	if (!gpio) {
-		return NULL;
-	}
-
-	cc2520->gpios[CC2520_GPIO_IDX_CCA].pin = DT_INST_GPIO_PIN(0, cca_gpios);
-	gpio_pin_configure(gpio, cc2520->gpios[CC2520_GPIO_IDX_CCA].pin,
-			   GPIO_INPUT | DT_INST_GPIO_FLAGS(0, cca_gpios));
-	cc2520->gpios[CC2520_GPIO_IDX_CCA].dev = gpio;
-
-	/* SFD */
-	gpio = device_get_binding(DT_INST_GPIO_LABEL(0, sfd_gpios));
-	if (!gpio) {
-		return NULL;
-	}
-
-	cc2520->gpios[CC2520_GPIO_IDX_SFD].pin = DT_INST_GPIO_PIN(0, sfd_gpios);
-	gpio_pin_configure(gpio, cc2520->gpios[CC2520_GPIO_IDX_SFD].pin,
-			   GPIO_INPUT | DT_INST_GPIO_FLAGS(0, sfd_gpios));
-	cc2520->gpios[CC2520_GPIO_IDX_SFD].dev = gpio;
-
-	/* FIFOP */
-	gpio = device_get_binding(DT_INST_GPIO_LABEL(0, fifop_gpios));
-	if (!gpio) {
-		return NULL;
-	}
-
-	cc2520->gpios[CC2520_GPIO_IDX_FIFOP].pin = DT_INST_GPIO_PIN(0, fifop_gpios);
-	gpio_pin_configure(gpio, cc2520->gpios[CC2520_GPIO_IDX_FIFOP].pin,
-			   GPIO_INPUT | DT_INST_GPIO_FLAGS(0, sfd_gpios));
-	cc2520->gpios[CC2520_GPIO_IDX_FIFOP].dev = gpio;
-
-	return cc2520->gpios;
-}
-
-
-static inline int configure_spi(const struct device *dev)
-{
-	struct cc2520_context *cc2520 = dev->data;
-
-	cc2520->spi = device_get_binding(DT_INST_BUS_LABEL(0));
-	if (!cc2520->spi) {
-		LOG_ERR("Unable to get SPI device");
-		return -ENODEV;
-	}
-
-#if DT_INST_SPI_DEV_HAS_CS_GPIOS(0)
-	cs_ctrl.gpio_dev = device_get_binding(
-		DT_INST_SPI_DEV_CS_GPIOS_LABEL(0));
-	if (!cs_ctrl.gpio_dev) {
-		LOG_ERR("Unable to get GPIO SPI CS device");
+	const struct cc2520_config *cfg = dev->config;
+
+	if (!device_is_ready(cfg->vreg_en.port) ||
```

Comment:
Maybe let's change this conditional to something like that?
<pre>
    const struct device *devs[] = {cfg->vreg_en.port, cfg->reset.port, cfg->fifo.port,
                                    cfg->cca.port, cfg->sfd.port, cfg->fifop.port};

    for (int i=0; i < ARRAY_SIZE(devs); i++) {
        if (!device_is_ready(devs[i]))
            return -ENODEV;
    }</pre>

Reply:
I think current approach is more efficient, no need to define an array of pointers to just iterate once over them

Reply:
assembler would be even more efficient but IMO. readability is also important but it's a moot point ;)